Doc/whatsnew/3.10.rst

```diff
@@ -249,3 +249,6 @@ Removed
 * Removed ``PyUnicode_AsUnicodeCopy()``. Please use :c:func:`PyUnicode_AsUCS4Copy` or
   :c:func:`PyUnicode_AsWideCharString`
   (Contributed by Inada Naoki in :issue:`41103`.)
+
+* Removed docs for the deprecated and removed ``symbol.sym_name``.
```

Comment:
I think that there should be one entry for the removal of the symbol name and its doc.

Reply:
I did not either.  Considering only symbol.h, this entry should say something like 
"Removed :module:`symbol`.  (Contributed by Lysandros Nikolaou in :issue: `40939` and Joannah Nanjekye in :issue:`41234`."  (Note: I am guessing at the :module: role.  Perhaps is presence is a mistake when the module and doc is removed.  Removing the test and docs for a module and other references in the docs are part of removing a module.  Finding the latter is non-trivial.)

However, this entry is currently in the C-API section.  It belongs in the currently empty Removed section a couple of section after Improved Modules.

The bigger problem is that the removal of symbol is a small part of the removal of the old parser and associated stuff done in 40939, none of which appears in What's New.  I believe other modules were removed. So it would be good if you expanded this issue and went through that issue, listed all the removed modules (I don't think that removed tests need to be listed), removed all the associated docs still present, and wrote one module removal entry.  The contributed part might only need the addition of "Pablo Galindo and ".

"Removed modules associated with the old parser: <list> (Contributed ...)

A followup would be to do the same at least for public C-API removals.

Reply:
IMO, I think I found this as sort of unfinished business of a former removal of things i.e we have docs for a feature that was removed already which is misleading. I have not bothered to look into the parser removal details though. I have done the required edit in the news entry.